simpleviewinc/tap-events-force
# EVF Button Changes

**Ticket**: [ZEN-556](https://jira.simpleviewtools.com/browse/ZEN-556)

## Context
* See the [Tech Spec here](https://docs.google.com/document/d/1oTOhGc1fpG0VhqXTq4ZumceZWoi1ln17wFVxG9SmlDE/edit)
* Events Force is stating that they need more customization over the Button within the sessions component
* Based on their explanation, they are having issues with styling it properly
* They would like to follow a process similar to the updates we did with the modal component, as seen in these tickets
  * https://jira.simpleviewtools.com/browse/ZEN-446
  * https://jira.simpleviewtools.com/browse/ZEN-502
* Basically, they would pass us a Button component, and we would add some logic to it, and pass it back to them


## Goal

* Allow the evf-button to work like the Modal component and be passed in externally
* Update all references to the button to use this custom evf-button instead of the current button
* Like the modal component, use the bootstrap button when in a development environment
* It should not load the bootstrap button in the production build
* Clean up the button styles


## Updates
* Added Button Component prop
  * Allows passing in a custom button component
  * Updated the modal context to be a Components context
    * Now stores both modal and button components
  * Rewrote button styles
* Added buttonType prop to evf button matching the Tech-Spec requirements
  * Allowed types are `selectSession`, `modalPrimary`, `modalSecondary`

## Testing

* Run the docker packages
  * EVF -  `keg evf pack run evf:zen-556-evf-button-changes`
  * Consumer - `keg consumer pack run consumer:zen-556-evf-button-changes`

> **Notes**
> Please test on both images.
> There were a lot of changes done to keg-hub and this is the first build since those changes were merged

* **Important**
  * There may be some cases where the button does not look correct
  * This is because the root button does not have any styles
  * The EU team is handling that
    * One example of this is the button height in the modal compared to the height in the session list
    * The EU team now controls this
  * We are only responsible for the children

**Test 1**
* Validate that the button children still work as expected
  * The buttons children should still look and act the same
  * The Root Button component should be using the reactstrap button component
* Updates were made to the modal provider
  * So ensure the modal component still displays and closes as expected

**Test 2**
  * Ensure the button icons still render as expected
    * Check for proper alignment
    * Check that they still display for each edge-case defined in the booking state
    * Check both check-mark and the Digit icon variations

**Test 3**
* Ensure all button actions still work as expected
  * Ensure the pending spinner icon still renders as expected
    * When you click book selected, the spinner should display and rotate
  * Modal cancel / book selected
  * Update the mock data to be some variation of single attendee booking state
    * Ensure the check-mark icon still displays when selecting a session
